cscs-checks/microbenchmarks/likwid/likwid.py

```diff
@@ -0,0 +1,176 @@
+
+        self.sanity_patterns = sn.assert_ge(bw_pattern, 0.0)
+        self.perf_patterns = {
+            'bandwidth': bw_pattern
+        }
+
+    def setup(self, partition, environ, **job_opts):
+        super().setup(partition, environ, **job_opts)
+
+        self.postbuild_cmd = ['make install PREFIX=%s INSTALL_CHOWN='
+                              '\'-g csstaff -o sebkelle\'' % self.stagedir]
+
+
+@rfm.required_version('>=2.16-dev0')
+@rfm.parameterized_test(*[[l,k] for l in ['L1', 'L2', 'L3']
+                        for k in ['load_avx', 'store_avx']],
+                        ['memory', 'load_avx'],
+                        ['memory', 'store_mem_avx'])
+class CPUBandwidth(BandwidthBase):
+    def __init__(self, mem_level, kernel_name):
+        super().__init__()
+
+        self.descr = 'CPU <- %s %s benchmark' % (mem_level, kernel_name)
+
+        self.valid_systems = ['daint:mc', 'daint:gpu', 'dom:gpu', 'dom:mc']
+
+        # the kernel to run in likwid
+        self.kernel_name = kernel_name
+        self.ml = mem_level
+
+        self.refs = {
+            'mc':  {
+                'load_avx': {'L1': 5100000, 'L2': 2000000, 'L3': 900000,
+                             'memory': 130000},
+                'store_avx': {'L1': 2800000, 'L2': 900000, 'L3': 480000},
+                'store_mem_avx': {'memory': 85000},
+            },
+            'gpu': {
+                'load_avx': {'L1': 2100000, 'L2': 850000, 'L3': 360000,
+                             'memory': 65000},
+                'store_avx': {'L1': 1200000, 'L2': 340000, 'L3': 210000},
+                'store_mem_avx': {'memory': 42500},
+            }
+        }
+        ref_proxy = {part: self.refs[part][kernel_name][mem_level]
+                     for part in self.refs.keys()}
+
+        self.reference = {
+            'daint:gpu': {
+                'bandwidth': (ref_proxy['gpu'], -0.1, None, 'MB/s')
+            },
+            'daint:mc': {
+                'bandwidth': (ref_proxy['mc'], -0.1, None, 'MB/s')
+            },
+            'dom:gpu': {
+                'bandwidth': (ref_proxy['gpu'], -0.1, None, 'MB/s')
+            },
+            'dom:mc': {
+                'bandwidth': (ref_proxy['mc'], -0.1, None, 'MB/s')
+            },
+        }
+
+    def setup(self, partition, environ, **job_opts):
+        self.data_size = self.system_cache_sizes[partition.fullname][self.ml]
+
+        self.num_cpus_per_task = self.system_num_cpus[partition.fullname]
+        numa_domains = self.system_numa_domains[partition.fullname]
+
+        num_cpu_dom = self.num_cpus_per_task / (len(numa_domains) *
+                                                self.num_tasks_per_core)
+        # result for daint:mc: '-w S0:100MB:18:1:2 -w S1:100MB:18:1:2'
+        # format: -w domain:data_size:nthreads:chunk_size:stride
+        # chunk_size and stride affect which cpus from <domain> are selected
+        workgroups = ' '.join(['-w %s:%s:%d:1:2' %
```

Comment:
`executable_opts` is a list so you can avoid the join here. Just create a list of the `-w blah:blah` arguments.